Src/Newtonsoft.Json/Linq/JArray.cs

```diff
@@ -176,9 +176,9 @@ public new static JArray Parse(string json, JsonLoadSettings settings)
             {
                 JArray a = Load(reader, settings);
 
-                if (reader.Read() && reader.TokenType != JsonToken.Comment)
+                while (reader.Read())
                 {
-                    throw JsonReaderException.Create(reader, "Additional text found in JSON string after parsing content.");
+                    // Any content encountered here other than a comment will throw in the reader.
```

Comment:
As became apparent. There's no way to actually detect `reader.TokenType != JsonToken.Comment` without `catch`ing the exception from the reader itself, and little point since the exception thrown is applicable.